Tasks/Common/ios-signing-common/ios-signing-common.ts

```diff
@@ -186,6 +186,95 @@ export async function getProvisioningProfileUUID(provProfilePath: string) {
     }
 }
 
+
+/**
+ * Find the type of the provisioning profile - development, app-store or ad-hoc
+ * @param provProfilePath
+ * @returns {string} type
+ */
+export async function getProvisioningProfileType(provProfilePath: string) {
+    var provProfileType: string;
+    try {
+        //find the provisioning profile details
+        var provProfileDetails:string;
+        var getProvProfileDetailsCmd:ToolRunner = tl.tool(tl.which('security', true));
+        getProvProfileDetailsCmd.arg(['cms', '-D', '-i', provProfilePath]);
+        getProvProfileDetailsCmd.on('stdout', function (data) {
+            if (data) {
+                if (provProfileDetails) {
+                    provProfileDetails = provProfileDetails.concat(data.toString().trim().replace(/[,\n\r\f\v]/gm, ''));
+                } else {
+                    provProfileDetails = data.toString().trim().replace(/[,\n\r\f\v]/gm, '');
+                }
+            }
+        })
+        await getProvProfileDetailsCmd.exec();
+
+        if (provProfileDetails) {
+            //write the provisioning profile to a plist
+            var tmpPlist = '_xcodetasktmp.plist';
+            fs.writeFileSync(tmpPlist, provProfileDetails);
+        } else {
+            throw tl.loc('ProvProfileDetailsNotFound', provProfilePath);
+        }
+
+        //get ProvisionsAllDevices - this will exist for enterprise profiles
+        var provisionsAllDevices: string = await printFromPlist('ProvisionsAllDevices', tmpPlist);
+        tl.debug('provisionsAllDevices = ' + provisionsAllDevices);
+        if(provisionsAllDevices && provisionsAllDevices.toLowerCase() === 'true') {
+            //ProvisionsAllDevices = true in enterprise profiles
+            provProfileType = 'enterprise';
+        } else {
+            var getTaskAllow: string = await printFromPlist('Entitlements:get-task-allow', tmpPlist);
+            tl.debug('getTaskAllow = ' + getTaskAllow);
+            if (getTaskAllow && getTaskAllow.trim().toLowerCase() === 'true') {
+                //get-task-allow = true means it is a development profile
+                provProfileType = 'development';
+            } else {
+                var provisionedDevices:string = await printFromPlist('ProvisionedDevices', tmpPlist);
+                if (!provisionedDevices) {
+                    // no provisioned devices for non-development profile means it is an app-store profile
+                    provProfileType = 'app-store';
+                } else {
+                    // non-development profile with provisioned devices - use ad-hoc
+                    provProfileType = 'ad-hoc';
+                }
+            }
+        }
+
+        //delete the temporary plist file
+        var deletePlistCommand:ToolRunner = tl.tool(tl.which('rm', true));
+        deletePlistCommand.arg(['-f', tmpPlist]);
+        await deletePlistCommand.exec();
```

Comment:
I will take a note of this for next update since this is not the only place we could switch. Also if we use fs.unlinkSync(), we will need to mock it for L0 tests.